pkg/controller/datavolume-controller.go

```diff
@@ -2958,8 +2959,9 @@ func (r *DatavolumeReconciler) getSizeFromPod(sourcePvc *corev1.PersistentVolume
 	}
 
 	pod, err := r.getOrCreateSizeDetectionPod(sourcePvc, dv)
-	if err != nil {
-		return 0, handleFailedPod(err, sizeDetectionPodName(sourcePvc), sourcePvc, r.recorder, r.client)
+	// Check if pod has failed and, in that case, record an event with the error
+	if podErr := handleFailedPod(err, sizeDetectionPodName(sourcePvc), sourcePvc, r.recorder, r.client); podErr != nil {
+		return 0, podErr
 	} else if !isPodComplete(pod) {
```

Comment:
you dont need else if you return in the if

Reply:
Sorry, I forgot to comment back. I can change this if it's preferable, but I kinda like how the error handling of the pod is concentrated in a single `if/else` block. Though it's not really necessary, I feel like the `if else` allows concentrating similar behavior in the same statement and makes it more readable.

Reply:
On the other hand, this guard statement is a common idiom in go:
```
if err := xyz(); err != nil {
   return err
}
```
and after spending some time reading `go`, I can say it is easy to read and reason about. It is obvious that in bad case if immediately returns, while in the good case it continues with next statements.

When I see `if err / else ` I expect some kind of complex logic around error handling, or some alternative behavior based on few conditions.

In this case both versions are correct. We might argue which way is more idiomatic.